kotlin/src/main/java/app/rive/runtime/kotlin/RiveSurfaceView.kt

```diff
@@ -49,17 +73,26 @@ abstract class RiveSurfaceView(context: Context, attrs: AttributeSet? = null) :
         super.onAttachedToWindow()
         // Register this SurfaceView for the SurfaceHolder callbacks below
         holder.addCallback(this)
+        isRunning = true
     }
 
     @CallSuper
     override fun surfaceCreated(holder: SurfaceHolder) {
         cppInit(activity, refreshPeriodNanos)
         renderer.setSurface(holder.surface)
         renderer.start()
+        isRunning = true
     }
 
     override fun onDetachedFromWindow() {
         super.onDetachedFromWindow()
         renderer.cleanup()
+        isRunning = false
+    }
+    override fun surfaceDestroyed(holder: SurfaceHolder) {
```

Comment:
I've been adding the `@CallSuper` decorator to force implementers to call this if they're overriding it. We might wanna do the same here

Reply:
Also, could there be a race-condition here between `onDetachedFromWindow()` and `onDestroyed()`?

Reply:
> oh no idea.. i thought i was just moving the call from the RiveAnimationView into the RiveSurfaceView, do you think that change could introduce a race condition?

I was just curious if there was a case where `onDetachedFromWindow()` wasn't called before/after `surfaceDestroyed()`